### PR TITLE
Updated scaledobject definition to use triggerAuthentication

### PR DESCRIPTION
**Pull request by therahulbhati**

### CODE REVIEW

**Comment by therahulbhati** [comment]
@jeffhollan @tomkerkhove Can you please review this PR?

**Comment by tomkerkhove** [comment]
I'm not a Go expert but maybe @ahmelsayed and @zroubalik can help!

File: deploy/deploy-consumer.yaml [hunk 1]
```diff
@@ -22,16 +22,13 @@ spec:
...
     spec:
       containers:
-      - name: rabbitmq-consumer
-        image: jeffhollan/rabbitmq-client:dev
-        imagePullPolicy: Always
-        command:
-          - receive
-        args:
-          - 'amqp://user:PASSWORD@rabbitmq.default.svc.cluster.local:5672'
-        envFrom:
```
Comment (tomkerkhove) [inline deploy/deploy-consumer.yaml:32]: If you remove this, then the consumer won't work anymore AFAIK

File: deploy/deploy-consumer.yaml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 apiVersion: v1
 kind: Secret
 metadata:
-  name: rabbitmq-consumer
+  name: rabbitmq-consumer-secret
 data:
-  RabbitMqHost: YW1xcDovL3VzZXI6UEFTU1dPUkRAcmFiYml0bXEuZGVmYXVsdC5zdmMuY2x1c3Rlci5sb2NhbDo1Njcy
+  host: YW1xcDovL3VzZXI6UEFTU1dPUkRAcmFiYml0bXEuZGVmYXVsdC5zdmMuY2x1c3Rlci5sb2NhbDo1Njcy
```
Comment (tomkerkhove) [inline deploy/deploy-consumer.yaml]: Let's revert and update the TriggerAuthentication
```suggestion
  RabbitMqHost: YW1xcDovL3VzZXI6UEFTU1dPUkRAcmFiYml0bXEuZGVmYXVsdC5zdmMuY2x1c3Rlci5sb2NhbDo1Njcy
```

File: deploy/deploy-consumer.yaml [hunk 3]
```diff
@@ -43,13 +40,26 @@ metadata:
...
+apiVersion: keda.k8s.io/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: rabbitmq-consumer-trigger
+  namespace: default
+spec:
+  secretTargetRef:
+    - parameter: host
+      name: rabbitmq-consumer-secret
+      key: host
```
Comment (tomkerkhove) [inline deploy/deploy-consumer.yaml]:
```suggestion
      key: RabbitMqHost
```

File: deploy/deploy-consumer.yaml [hunk 4]
```diff
@@ -22,16 +22,13 @@ spec:
...
-        envFrom:
-        - secretRef:
-            name: rabbitmq-consumer
+        - name: rabbitmq-consumer
+          image: jeffhollan/rabbitmq-client:dev
+          imagePullPolicy: Always
+          command:
+            - receive
+          args:
+            - "amqp://user:PASSWORD@rabbitmq.default.svc.cluster.local:5672"
```
Comment (tomkerkhove) [inline deploy/deploy-consumer.yaml:31]: Let's bring this back
```suggestion
            - "amqp://user:PASSWORD@rabbitmq.default.svc.cluster.local:5672"
        envFrom:
        - secretRef:	
            name: rabbitmq-consumer-secret
```

**Review by tomkerkhove — CHANGES_REQUESTED** [review]

**Comment by tomkerkhove** [comment]
Didn't notice it was just for the YAML spec, sorry @ahmelsayed @zroubalik !

@therahulbhati I've added some comments as it won't work in its current state given the deployment doesn't have a connection anymore.

File: deploy/deploy-consumer.yaml (hunk 1 above)
Comment (therahulbhati) [reply deploy/deploy-consumer.yaml:32]: Yes, you are right

File: deploy/deploy-consumer.yaml (hunk 4 above)
Comment (therahulbhati) [reply deploy/deploy-consumer.yaml:31]: We are not using secretRef in consumer. Consumer is getting url from args. so this shouldn't be a problem

File: deploy/deploy-consumer.yaml (hunk 1 above)
Comment (therahulbhati) [reply deploy/deploy-consumer.yaml:32]: I would like to correct myself, this won't cause any problem. Since consumer is getting url from args

**Comment by therahulbhati** [comment]
@tomkerkhove I made changes as you suggested expect secretRef one because the consumer never used that secretRef, it was reading URL from args.

**Review by tomkerkhove — APPROVED** [review]

**Comment by tomkerkhove** [comment]
> @tomkerkhove I made changes as you suggested expect secretRef one because the consumer never used that secretRef, it was reading URL from args.

This was probably because we didn't have trigger auth yet and it required an env var for that.

Thanks!

**Pull request merged**